apps/travelperk/actions.py

```diff
@@ -204,42 +211,42 @@ def create_invoice_lineitems(org_id, invoice, expense, user_role, amount):
         object: Created expense object.
     """
 
-    # Determine the employee's email based on the user role
+    # Establish a connection to the Fyle platform
     platform_connection = create_fyle_connection(org_id)
+    matched_transaction_found = []
 
+    # Determine the employee's email based on the user role
     if user_role in ['TRAVELLER', 'BOOKER']:
         travelperk_employee = getattr(expense, ROLE_EMAIL_MAPPING.get(user_role, None))
         employee_email = get_employee_email(platform_connection, travelperk_employee)
     else:
-        employee_email = get_email_from_credit_card(platform_connection, expense.credit_card_last_4_digits)
+        employee_email, matched_transaction_found = get_email_from_credit_card(platform_connection, expense, amount)
 
-    # Create the payload for the expense
-    payload = construct_expense_payload(org_id, expense, amount)
+    file_ids = construct_file_ids(platform_connection, invoice.pdf)
+
+    if not len(matched_transaction_found):
```

Comment:
```suggestion
    if not matched_transaction_found:
```
len() is not required